src/lib/plot/bar.typ

```diff
@@ -52,19 +52,30 @@
     let right = left + w
     let width = (right - left) / len
 
+
     if self.mode in ("basic", "clustered") {
       left = left + width * n
       right = left + width
     }
-
     if (left <= x-axis.max and right >= x-axis.min and
         y-min <= y-axis.max and y-max >= y-axis.min) {
       left = calc.max(left, x-axis.min)
       right = calc.min(right, x-axis.max)
       y-min = calc.max(y-min, y-axis.min)
       y-max = calc.min(y-max, y-axis.max)
 
-      draw.rect((left, y-min), (right, y-max))
+
+      draw.group(ctx => {
+        draw.rect((left, y-min), (right, y-max))
+        if ctx.style.data-label != none {
+          let offset = ctx.style.data-label.at("offset")
```

Comment:
Style data must come from `self.style`. This allows setting per-bar/plot style data.